packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx

```diff
@@ -1,436 +1,215 @@
 import * as React from 'react';
 import {
   BaseComponent,
-  autobind,
-  buttonProperties,
-  anchorProperties,
   css,
-  divProperties,
   getId,
-  getNativeProps
-} from '../../Utilities';
-import { ICommandBar, ICommandBarProps } from './CommandBar.Props';
-import { FocusZone, FocusZoneDirection } from '../../FocusZone';
-import { ContextualMenu, IContextualMenuProps, IContextualMenuItem, hasSubmenuItems } from '../../ContextualMenu';
-import { DirectionalHint } from '../../common/DirectionalHint';
-import {
-  Icon,
-  IIconProps
-} from '../../Icon';
-import { FontClassNames } from '../../Styling';
+  autobind
+} from 'office-ui-fabric-react/lib/Utilities';
+import { ICommandBar, ICommandBarProps, ICommandBarItemProps } from './CommandBar.Props';
+import { CommandBarButton } from 'office-ui-fabric-react/lib/Button';
+import { OverflowSet, IOverflowSet } from 'office-ui-fabric-react/lib/OverflowSet';
+import { ResizeGroup } from 'office-ui-fabric-react/lib/ResizeGroup';
+import { TooltipHost } from 'office-ui-fabric-react/lib/Tooltip';
 import * as stylesImport from './CommandBar.scss';
-const styles: any = stylesImport;
 
-const OVERFLOW_KEY = 'overflow';
-const OVERFLOW_WIDTH = 41.5;
+// tslint:disable-next-line:no-any
+const styles: any = stylesImport;
 
-export interface ICommandBarState {
-  renderedItems?: IContextualMenuItem[];
-  renderedOverflowItems?: IContextualMenuItem[];
-  expandedMenuItemKey?: string;
-  expandedMenuId?: string;
-  contextualMenuProps?: IContextualMenuProps;
-  contextualMenuTarget?: HTMLElement;
-  renderedFarItems?: IContextualMenuItem[];
+export interface ICommandBarData {
```

Comment:
Why not put this in the props file? As it is, you have a circular dependency between this file and the props file.